packages/app/src/cli/services/context/identifiers.ts

```diff
@@ -33,6 +33,8 @@ export interface LocalSource {
   graphQLType: string
   type: string
   handle: string
+  configuration: {name: string}
+  isConfigExtension: boolean
```

Comment:
I see where you are using `isConfigExtension`, but where are you using `configuration: {name: string}`?

Reply:
🤔 Nice find. It looks like it wasn't being used. The `configuration` object has been deleted from `LocalSource` in bbdd8656ce84dcff3886ade7fa49fac7b6713f39. 

<details>

<summary>Search results for LocalSource</summary>

```
80 results - 10 files

packages/app/src/cli/services/context/identifiers.ts:
  29  
  30  
  31: export interface LocalSource {
  31    localIdentifier: string
  32    localIdentifier: string

packages/app/src/cli/services/context/prompts.test.ts:
    1  import {deployConfirmationPrompt, extensionMigrationPrompt, matchConfirmationPrompt, SourceSummary} from './prompts.js'
    2: import {RemoteSource, LocalSource} from './identifiers.js'
    3  import {IdentifiersExtensions} from '../../models/app/identifiers.js'

  342      const [localIdentifier, graphQLType, type, handle, isConfigExtension] = ['', '', '', 'foo', false]
  343:     const localSource = {localIdentifier, graphQLType, type, handle, isConfigExtension, configuration: {name: ''}}
  344      const remoteSource = {uuid: '', type: '', id: '', title: 'bar'}

  346      // When
  347:     const response = await matchConfirmationPrompt(localSource, remoteSource)
  348  

  439    identifiers?: IdentifiersExtensions
  440:   toCreate?: LocalSource[]
  441    onlyRemote?: RemoteSource[]

packages/app/src/cli/services/context/prompts.ts:
    1: import {LocalSource, RemoteSource} from './identifiers.js'
    2  import {LocalRemoteSource} from './id-matching.js'

   16  export async function matchConfirmationPrompt(
   17:   local: LocalSource,
   18    remote: RemoteSource,

   28  export async function selectRemoteSourcePrompt(
   29:   localSource: LocalSource,
   30    remoteSourcesOfSameType: RemoteSource[],

   38    const uuid = await renderAutocompletePrompt({
   39:     message: `How would you like to deploy your "${localSource.handle}"?`,
   40      choices: remoteOptions,

   48    identifiers: IdentifiersExtensions
   49:   toCreate: LocalSource[]
   50    dashboardOnly: RemoteSource[]

  111    localRegistration: IdentifiersExtensions,
  112:   toCreate: LocalSource[],
  113    dashboardOnly: RemoteSource[],

  175    localRegistration: IdentifiersExtensions,
  176:   toCreate: LocalSource[],
  177    dashboardOnly: RemoteSource[],

packages/app/src/cli/services/context/identifiers-extensions.ts:
    2  import {automaticMatchmaking} from './id-matching.js'
    3: import {EnsureDeploymentIdsPresenceOptions, LocalSource, MatchingError, RemoteSource} from './identifiers.js'
    4  import {deployConfirmationPrompt, extensionMigrationPrompt, matchConfirmationPrompt} from './prompts.js'

  122  
  123: async function createExtensions(extensions: LocalSource[], appId: string) {
  124    const token = await ensureAuthenticatedPartners()

packages/app/src/cli/services/dev/migrate-to-ui-extension.ts:
   5  } from '../../api/graphql/extension_migrate_to_ui_extension.js'
   6: import {LocalSource, RemoteSource} from '../context/identifiers.js'
   7  import {IdentifiersExtensions} from '../../models/app/identifiers.js'

  14  export function getUIExtensionsToMigrate(
  15:   localSources: LocalSource[],
  16    remoteSources: RemoteSource[],

  18  ) {
  19:   const ids = getExtensionIds(localSources, identifiers)
  20    const remoteExtensionTypesToMigrate = ['CHECKOUT_UI_EXTENSION', 'CUSTOMER_ACCOUNTS_UI_EXTENSION']
  21  
  22:   return localSources.reduce<LocalRemoteSource[]>((accumulator, localSource) => {
  23:     if (localSource.type === 'ui_extension') {
  24        const remoteSource = remoteSources.find((source) => {
  25:         const matchesId = source.uuid === ids[localSource.localIdentifier]
  26:         const matchesTitle = slugify(source.title) === slugify(localSource.handle)
  27  

  37        if (typeIsToMigrate) {
  38:         accumulator.push({local: localSource, remote: remoteSource})
  39        }

packages/app/src/cli/services/dev/migrate-to-ui-extension.test.ts:
   1  import {getUIExtensionsToMigrate, migrateExtensionsToUIExtension} from './migrate-to-ui-extension.js'
   2: import {LocalSource, RemoteSource} from '../context/identifiers.js'
   3  import {ExtensionMigrateToUiExtensionQuery} from '../../api/graphql/extension_migrate_to_ui_extension.js'

  10  
  11: function getLocalExtension(attributes: Partial<LocalSource> = {}) {
  12    return {

  19      ...attributes,
  20:   } as unknown as LocalSource
  21  }

packages/app/src/cli/services/context/id-matching.ts:
    1: import {RemoteSource, LocalSource} from './identifiers.js'
    2  import {IdentifiersExtensions} from '../../models/app/identifiers.js'

    8  export interface LocalRemoteSource {
    9:   local: LocalSource
   10    remote: RemoteSource

   15    toConfirm: LocalRemoteSource[]
   16:   toCreate: LocalSource[]
   17:   toManualMatch: {local: LocalSource[]; remote: RemoteSource[]}
   18  }

   22   */
   23: const sameTypeAndName = (local: LocalSource, remote: RemoteSource) => {
   24    return remote.type === local.graphQLType && slugify(remote.title) === slugify(local.handle)

   32  function matchByNameAndType(
   33:   local: LocalSource[],
   34    remote: RemoteSource[],
   35    remoteIdField: 'id' | 'uuid',
   36: ): {matched: IdentifiersExtensions; pending: {local: LocalSource[]; remote: RemoteSource[]}} {
   37    const uniqueLocal = uniqBy(local, (elem) => [elem.graphQLType, elem.handle])

   41  
   42:   uniqueLocal.forEach((localSource) => {
   43:     const possibleMatch = uniqueRemote.find((remoteSource) => sameTypeAndName(localSource, remoteSource))
   44:     if (possibleMatch) validMatches[localSource.localIdentifier] = possibleMatch[remoteIdField]
   45    })

   55    ids: IdentifiersExtensions,
   56:   localSources: LocalSource[],
   57    remoteSources: RemoteSource[],

   59    migrated: IdentifiersExtensions
   60:   pending: {local: LocalSource[]; remote: RemoteSource[]}
   61  } {

   78  
   79:   localSources
   80      .filter((extension) => extension.type === 'function')

   91  
   92:   const pendingLocal = localSources.filter((elem) => !migrated[elem.localIdentifier])
   93    const pendingRemote = remoteSources.filter((registration) => !Object.values(migrated).includes(registration.uuid))

  108  function matchByUniqueType(
  109:   localSources: LocalSource[],
  110    remoteSources: RemoteSource[],
  111  ): {
  112:   toCreate: LocalSource[]
  113:   toConfirm: {local: LocalSource; remote: RemoteSource}[]
  114:   pending: {local: LocalSource[]; remote: RemoteSource[]}
  115  } {
  116:   const localGroups = groupBy(localSources, 'graphQLType')
  117    const localUnique = Object.values(pickBy(localGroups, (group, key) => group.length === 1)).flat()

  121  
  122:   const toConfirm: {local: LocalSource; remote: RemoteSource}[] = []
  123:   const toCreate: LocalSource[] = []
  124  

  139    // it means that we need to create them.
  140:   const localDuplicated = difference(localSources, localUnique)
  141    const remotePending = difference(

  165  export async function automaticMatchmaking(
  166:   localSources: LocalSource[],
  167    remoteSources: RemoteSource[],

  170  ): Promise<MatchResult> {
  171:   const ids = getExtensionIds(localSources, identifiers)
  172    const localUUIDs = Object.values(ids)
  173  
  174:   const existsRemotely = (local: LocalSource) =>
  175      remoteSources.some(

  180      ids,
  181:     localSources.filter((local) => !existsRemotely(local)),
  182      remoteSources.filter((remote) => !localUUIDs.includes(remote[remoteIdField])),

  208  export function getExtensionIds(
  209:   localSources: LocalSource[],
  210    identifiers: IdentifiersExtensions,
  211  ): IdentifiersExtensions {
  212:   const localSourcesIds = localSources.map((source) => source.localIdentifier)
  213  
  214:   return pickBy(identifiers, (_, id) => localSourcesIds.includes(id))
  215  }

packages/app/src/cli/services/dev/migrate-flow-extension.test.ts:
   1  import {getFlowExtensionsToMigrate, migrateFlowExtensions} from './migrate-flow-extension.js'
   2: import {LocalSource, RemoteSource} from '../context/identifiers.js'
   3  import {MigrateFlowExtensionMutation} from '../../api/graphql/extension_migrate_flow_extension.js'

  10  
  11: function getLocalExtension(attributes: Partial<LocalSource> = {}) {
  12    return {

  18      ...attributes,
  19:   } as unknown as LocalSource
  20  }

packages/app/src/cli/services/dev/migrate-flow-extension.ts:
   1: import {LocalSource, RemoteSource} from '../context/identifiers.js'
   2  import {IdentifiersExtensions} from '../../models/app/identifiers.js'

  14  export function getFlowExtensionsToMigrate(
  15:   localSources: LocalSource[],
  16    remoteSources: RemoteSource[],

  18  ) {
  19:   const ids = getExtensionIds(localSources, identifiers)
  20    const localExtensionTypesToMigrate = ['flow_action', 'flow_trigger']

  26  
  27:   const local = localSources.filter((source) => localExtensionTypesToMigrate.includes(source.type))
  28    const remote = remoteSources.filter((source) => remoteExtensionTypesToMigrate.includes(source.type))

  36  
  37:   return local.reduce<LocalRemoteSource[]>((accumulator, localSource) => {
  38:     const localSourceId = ids[localSource.localIdentifier] ?? 'unknown'
  39:     const remoteSource = remoteSourcesMap.get(localSourceId) || remoteSourcesMap.get(localSource.localIdentifier)
  40:     const typeMatch = typesMap.get(localSource.type) === remoteSource?.type
  41  
  42      if (remoteSource && typeMatch) {
  43:       accumulator.push({local: localSource, remote: remoteSource})
  44      }

packages/app/src/cli/services/context/id-manual-matching.ts:
   1  import {selectRemoteSourcePrompt} from './prompts.js'
   2: import {LocalSource, RemoteSource} from './identifiers.js'
   3  import {IdentifiersExtensions} from '../../models/app/identifiers.js'

   6    identifiers: IdentifiersExtensions
   7:   toCreate: LocalSource[]
   8    onlyRemote: RemoteSource[]

  24    options: {
  25:     local: LocalSource[]
  26      remote: RemoteSource[]
```
</details>